BUILDING.md

```diff
@@ -0,0 +1,246 @@
+- [Gentoo](#building-cherrytree-on-gentoo)
+- [Fedora](#building-cherrytree-on-fedora)
+- [Opensuse](#building-cherrytree-on-opensuse)
+- [MacOs](#building-cherrytree-on-macos)
+- [Windows](#building-cherrytree-on-windows)
+
+
+## Build/Debug with Visual Studio Code on Linux
+https://code.visualstudio.com/docs/setup/linux
+required installation of Extension "C/C++"
+```sh
+cd cherrytree
+code .
+```
+Build with: Ctrl+Shift+B
+Debug with: F5
+
+## Build/Debug with Visual Studio Code using a container
+It is possible to use a container as a full-featured development environment from VS Code.
+This works on any operating system that supports Docker.
+
+1. Install the [system requirements](https://code.visualstudio.com/docs/remote/containers#_system-requirements).
+2. Open the project in VS Code.
+3. (optional) Edit `.devcontainer/devcontainer.json` to set your `DISPLAY` environment variable and/or edit other settings.
+   It is also possible to run the container on a remote Docker host, see the comment at the end.
+4. Run the *Remote-Containers: Open Folder in Container...* command.
+5. See previous section for Build and Debug instructions.
+
+## To build using the bundled spdlog and fmt libraries
+```sh
+./build.sh bundledspdfmt
+```
+
+## To create an AppImage bundle
+```sh
+./build.sh appimage
+```
+
+## To generate a backtrace for a crash bug report
+```sh
+./build.sh debug
+gdb ./build/cherrytree
+(gdb) r
+```
+...after reproducing the crash
+```sh
+(gdb) bt
+```
+
+## Building Cherrytree on Ubuntu (+Linux Mint/Ubuntu)
+Install dependencies:
+```sh
+sudo apt install build-essential cmake ninja-build libgtkmm-3.0-dev libgtksourceviewmm-3.0-dev libxml++2.6-dev libsqlite3-dev gettext libgspell-1-dev libcurl4-openssl-dev libuchardet-dev libfribidi-dev libvte-2.91-dev libfmt-dev libspdlog-dev
+sudo apt install texlive-latex-base dvipng # optional for LatexBoxes support
+```
+Note: On Debian10 / Ubuntu 18.04 libfmt-dev and libspdlog-dev are not used since too old; bundled source code is built instead
+Get cherrytree source, compile and run:
+```sh
+git clone https://github.com/giuspen/cherrytree.git
+cd cherrytree
+git submodule update --init
+./build.sh
+./build/cherrytree
+```
+
+To create a debian package
+```sh
+./build.sh deb
+```
+
+Install documentation:
+```sh
+sudo apt install devhelp libgtkmm-3.0-doc libgtksourceviewmm-3.0-doc libglibmm-2.4-doc libpangomm-1.4-doc libxml++2.6-doc libgspell-1-doc libvte-2.91-doc
+```
+devhelp
+```sh
+xdg-open /usr/share/doc/libgtkmm-3.0-doc/reference/html/index.html
+xdg-open /usr/share/doc/libgtksourceviewmm-3.0-doc/reference/html/index.html
+xdg-open /usr/share/doc/libglibmm-2.4-doc/reference/html/index.html
+xdg-open /usr/share/doc/libpangomm-1.4-doc/reference/html/index.html
+xdg-open /usr/share/doc/libxml++2.6-doc/reference/html/index.html
+xdg-open /usr/share/doc/libgspell-1-dev/html/index.html
+```
+
+## Building Cherrytree on Arch Linux (+Linux Mint/Ubuntu)
```

Comment:
This should be changed to "... on Arch Linux (+ Manjaro)" as Manjaro is based on Arch Linux... I guess a copy/paste bug